syntax = "proto3";

package order_control;

service OrderControl {}

message ItemRequestCreateOrder {
  int32 product_id = 1;
  int32 amount = 2;
}

message ItemResponseCreateOrder {
  int32 product_id = 1;
  int32 status = 2;
}

message RequestCreateOrder { repeated ItemRequestCreateOrder item_list = 1; }

message ResponseCreateOrder { repeated ItemResponseCreateOrder item_list = 1; }

// ------------------------------------------

message RequestCancelOrder { int32 order_id = 1; }

message ResponseCancelOrder { int32 status = 1; }

// ------------------------------------------

message RequestFinishExecution {}

message ResponseFinishExecution {
  int32 stock_control_response = 1;
  int32 number_of_existing_orders = 2;
}
